frontend/packages/dev-console/src/components/topology/TopologyEdgePanel.tsx

```diff
@@ -1,26 +1,35 @@
 import * as React from 'react';
+import { connect } from 'react-redux';
+import * as classNames from 'classnames';
+import * as _ from 'lodash';
 import { BaseEdge } from '@console/topology';
+import { RootState } from '@console/internal/redux';
+import { referenceFor, K8sResourceKind } from '@console/internal/module/k8s';
 import {
   ActionsMenu,
   ResourceLink,
   SidebarSectionHeading,
+  ExternalLink,
 } from '@console/internal/components/utils';
 import {
   TYPE_CONNECTS_TO,
   TYPE_EVENT_SOURCE_LINK,
   TYPE_REVISION_TRAFFIC,
   TYPE_SERVICE_BINDING,
+  TYPE_TRAFFIC_CONNECTOR,
 } from './const';
 import { edgeActions } from './actions/edgeActions';
-import * as classNames from 'classnames';
-import * as _ from 'lodash';
-import { referenceFor } from '@console/internal/module/k8s';
 import { TopologyDataModel, TopologyDataObject } from './topology-types';
+import { getKialiLink } from './topology-utils';
+
+interface StateProps {
+  consoleLinks?: K8sResourceKind[];
+}
 
-export type TopologyEdgePanelProps = {
+export interface TopologyEdgePanelProps extends StateProps {
   edge: BaseEdge;
   data: TopologyDataModel;
-};
+}
```

Comment:
You can extend `type`s just like `interface`s... the syntax just changes:

```
type StateProps = {};
type TopologyEdgePanelProps = {} & StateProps;
```

Reply:
Yeah but in most of the places we use interface so for consistency across files I used interface will change it to type

Reply:
I think interfaces were a poor knowledge decision originally. TypeScript has gone through a lot of changes... back in the day `type` wasn't available. Most of our stuff was classes before I believe, which implies `interfaces`. I don't think it really matters one way or another... but I favour `types` for all things type related and `interfaces` for class-based implementation... a la Java Interfaces.

Personally, I wouldn't change the type when modifying code, but either way, it's a minor nit.

Reply:
Got it